utils/resultwriter.go

```diff
@@ -210,7 +210,14 @@ func printMessage(message string) {
 	log.Output("💬" + message)
 }
 
-func GenerateSarifReportFromResults(results *Results, isMultipleRoots, includeLicenses bool, allowedLicenses []string) (report *sarif.Report, err error) {
+func filterAndPatchRunsIfRequired(requestedScans []SubScanType, subScan SubScanType, scanType CommandType, results *Results, scanResults []*sarif.Run) (filtered []*sarif.Run) {
+	if !shouldScannerBeCalled(requestedScans, subScan, scanType) {
+		return
+	}
+	return patchRunsToPassIngestionRules(subScan, results, scanResults...)
+}
+
+func GenerateSarifReportFromResults(results *Results, isMultipleRoots, includeLicenses bool, allowedLicenses []string, requestedScans []SubScanType, scanType CommandType) (report *sarif.Report, err error) {
```

Comment:
```suggestion
func GenerateSarifReportFromResults(results *Results, isMultipleRoots, includeLicenses bool, allowedLicenses []string, requestedScans []SubScanType) (report *sarif.Report, err error) {
```
no need for `scanType CommandType` since you have the value stored at `results *Results`

Reply:
gotcha